### DISCUSSION TITLE
Command-line option for tabs

### DISCUSSION THREAD

**Discussion by ricardo-reis-1970** [linuxdeepin/developer-center#10670] [Features Request & Ideas | 特性请求 & 头脑风暴]
I'm a fullstack dev, working on Ubuntu 24.04.

I found out about `deepin-terminal` yesterday, on a quest to replace ``gnome-terminal`, and I found it just beautiful. A few things intrigued me, such as Ctrl-Tabbing through tabs being RTL, or not having any fonts to chose from in the dropdown, but these are minor things.

A thing that I found lacking in this terminal is the ability to control its configuration via command-line options. I have many projects where I need to open several terminals on the same project, on different sub-folders: frontend (React, CSS), backend (Python, C, Rust, data/db...). Some projects really use quite a few different technologies. I currently do this with `gnome-terminal` via composing options like:

For each project, I need to open different tabs on different subfolders. If a subfolder is a Python project, it should activate the virtual environment. I have all this scripted for `gnome-terminal` with a script that essentially composes a command-line like this:
```
gnome-terminal [--tab --title <TITLE> --profile=default --working-directory=<DIRECTORY>]...
```

The full command-line is composed in a script either via some default folders or based on folder names on a config file local to each project. The reason I go for tabs instead of screen split is that my eyes are terrible and I need all the space I can get for big fonts.

But for now I could not find a way to script this with `deepin-terminal`. Obviously I could open a terminal and manually create these tabs, but I'm looking for a programmatic way that is repeatable, automatic, adjustable to each project, etc.

I thought that perhaps I could have individual config files on each project root and then just run the terminal pointing to these, like terminator does with its -g/--config option, but I could not find any reference to anything similar. The only config I could find was in ~/.config/terminology/config/standard/base.cfg, and this is a binary file.

Is any of this possible with `deepin-terminal`?

**Comment by pmzgit** [comment]
## **tmux** is a terminal multiplexer. It lets you switch easily between several programs in one terminal, detach them (they keep running in the background) and reattach them to a different terminal.

https://github.com/tmux/tmux/wiki
## Self-contained, pretty and versatile **.tmux.conf** configuration file.
https://github.com/gpakosz/.tmux

## File transfer in the tmux （aka： lrzsz）
https://github.com/trzsz/trzsz-ssh

**Reply by ricardo-reis-1970** [reply]
Dear @pmzgit,

Thank you very much! You did not have to go out of your way to assist.

 I tried `tmux`, but I was disappointed that it was much like `screen`. After seeing `deepin-terminal`, I'm not going back to that.

I started using these [ADM-5](https://terminals-wiki.org/wiki/index.php/Lear_Siegler_ADM-5) terminals back in 1989 and I paid my dues to the 80s.

In fact, I can use `deepin-terminal` just like I need, as long as I open one terminal and then manually add tabs and change directories. I was looking for a way to do this from the command line, so that I could script something like `gotowork <project folder>` and have 5 tabs open on different folders within the project, with different themes and all.

**Comment by BLumia** [comment]
> or not having any fonts to chose from in the dropdown

Which distro are you using currently? It should be visible if you are using deepin, and it was a known issue for older version of deepin-terminal but likely already addressed by @ArchieMeng few months ago if I remembered correctly.

For other features, also cc @ArchieMeng :)

**Reply by ricardo-reis-1970** [reply]
I am using Ubuntu 24.04 / Wayland. Until last Saturday, when I found `deepin-terminal` and then started browsing around Deepin's website, I had no idea that there was a Deepin Linux distro. Screenshots look amazing, by the way.

For the moment I'm stuck with Ubuntu, but I don't think the issue I reported has much to do with the distro, unless `deepin-terminal` comes with extra functionality when you're running the Deepin distro... Does it?

**Reply by ArchieMeng** [reply]
@ricardo-reis-1970 In short, basically you just want a new command line option to open separate tabs with different initialization scripts?

**Reply by ricardo-reis-1970** [reply]
@ArchieMeng, in short, yes!

I'd like to have for each tab (most of) the same options of the entire window: working directory, title, eventually init script / execute command.

Obviously not the `--window-mode` because that is for the outer window.

There is a `--session` option, but the usage of it is not clear to me, as there is no apparent way to save a session.

There is no option to select the theme of the terminal. I tried `--platformtheme ` with different themes from the list but nothing changed.

In short, I'm looking to start it in the state it woulg get to after clicking around several times. We that prefer terminals are not too keen on mouse.
